### ISSUE TITLE
[Bug] Galera Init job keeps failing

### ISSUE THREAD

**Issue by boedy** [mariadb-operator/mariadb-operator#656]
**Documentation**
- [x] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**
When deploying a galera cluster the `galera-init` jobs keeps failing. It seems that as a result of this failure, the cluster remains in a `intializing` status.

**Expected behaviour**
Cluster should initialize without any issues

**Steps to reproduce the bug**
Deploy cluster:

```yaml
apiVersion: k8s.mariadb.com/v1alpha1
kind: MariaDB
metadata:
  name: mariadb-galera
spec:
  storage:
    size: 1Gi

  replicas: 3

  podSecurityContext:
    runAsUser: 0

  galera:
    enabled: true

  service:
    type: ClusterIP

  primaryService:
    type: ClusterIP

  secondaryService:
    type: ClusterIP

  metrics:
    enabled: true
```

**Debug information**
init jobs logs
```
2024-05-29 18:53:02+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.11.7+maria~ubu2204 started
2024-05-29 18:53:02+00:00 [Warn] [Entrypoint]: /sys/fs/cgroup///memory.pressure not writable, functionality unavailable to MariaDB
2024-05-29 18:53:02+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-05-29 18:53:02+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.11.7+maria~ubu2204 started
2024-05-29 18:53:02+00:00 [Note] [Entrypoint]: Starting temporary server
2024-05-29 18:53:02+00:00 [Note] [Entrypoint]: Waiting for server startup
2024-05-29 18:53:02 0 [Note] Starting MariaDB 10.11.7-MariaDB-1:10.11.7+maria~ubu2204 source revision 87e13722a95af5d9378d990caf48cb6874439347 as process 60
2024-05-29 18:53:02 0 [Warning] You need to use --log-bin to make --binlog-format work.
2024-05-29 18:53:02 0 [ERROR] mariadbd: Can't lock aria control file '/var/lib/mysql/aria_log_control' for exclusive use, error: 11. Will retry for 30 seconds
2024-05-29 18:53:32 0 [ERROR] mariadbd: Got error 'Could not get an exclusive lock; file is probably in use by another process' when trying to use aria control file '/var/lib/mysql/aria_log_control'
2024-05-29 18:53:32 0 [ERROR] Plugin 'Aria' registration as a STORAGE ENGINE failed.
2024-05-29 18:53:32 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2024-05-29 18:53:32 0 [Note] InnoDB: Number of transaction pools: 1
2024-05-29 18:53:32 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2024-05-29 18:53:32 0 [Note] mariadbd: O_TMPFILE is not supported on /tmp (disabling future attempts)
2024-05-29 18:53:32 0 [Note] InnoDB: Initializing buffer pool, total size = 8.000GiB, chunk size = 128.000MiB
2024-05-29 18:53:32 0 [Note] InnoDB: Completed initialization of buffer pool
2024-05-29 18:53:32 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
2024-05-29 18:53:32 0 [ERROR] InnoDB: Unable to lock ./ibdata1 error: 11
2024-05-29 18:53:32 0 [Note] InnoDB: Check that you do not already have another mariadbd process using the same InnoDB data or log files.
2024-05-29 18:53:32 0 [ERROR] InnoDB: Plugin initialization aborted with error Generic error
2024-05-29 18:53:32 0 [Note] InnoDB: Starting shutdown...
2024-05-29 18:53:32 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2024-05-29 18:53:32 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-05-29 18:53:32 0 [ERROR] Could not open mysql.plugin table: "Unknown storage engine 'Aria'". Some plugins may be not loaded
2024-05-29 18:53:32 0 [ERROR] Failed to initialize plugins.
2024-05-29 18:53:32 0 [ERROR] Aborting
2024-05-29 18:53:33+00:00 [ERROR] [Entrypoint]: Unable to start server.
```

**Environment details**:
- Kubernetes version: 1.25
- Kubernetes distribution: k3s
- mariadb-operator version: v0.0.28
- Install method: helm

**Comment by boedy** [comment]
I attempted to replicate this issue on a managed Kubernetes cluster provided by DigitalOcean, running version v1.27.10. I was unable to reproduce the problem in that environment.

The affected cluster is running on bare metal and uses Linstor for its storage layer. I suspect that the issue may be related to this specific setup. It is also possible that clusters using Longhorn as their storage layer could experience a similar issue.

**Comment by boedy** [comment]
Turned out to be an issue with the ClusterRole permission not having the delete verb on the configmap resource, which prevented the operator from removing init job if the Galera cluster is a different namespace from the operator 🤦 . This seems to be already patched in helm release `0.28.1`

**Comment by mmontes11** [comment]
Sorry that you experienced this, and glad that you found the issue. Indeed, we had to release a patch version 0.28.1 to fix the ClusterRole
